src/test/FreeC/Pass/ExportPassTests.hs

```diff
@@ -0,0 +1,126 @@
+-- | This module contains tests for "FreeC.Pass.ExportPass".
+
+module FreeC.Pass.ExportPassTests
+  ( testExportPass
+  )
+where
+
+import           Test.Hspec
+
+import           Data.List                      ( find )
+import           Data.Maybe                     ( fromJust )
+import qualified Data.Set                      as Set
+
+import qualified FreeC.Backend.Coq.Syntax      as Coq
+import           FreeC.Environment
+import           FreeC.Environment.Entry
+import           FreeC.Environment.ModuleInterface
+import           FreeC.IR.Syntax               as IR
+import           FreeC.Monad.Class.Testable
+import           FreeC.Monad.Converter
+import           FreeC.Pass.ExportPass
+import           FreeC.Pretty
+import           FreeC.Test.Environment
+import           FreeC.Test.Parser
+
+-- | Looks up an exported entry in the given module interface.
+--   also checks if the entry is defined in the given module.
+--   Returns 'Nothing' if either the check or the lookup fails.
+lookupExportedEntry :: IR.Scope -> IR.QName -> ModuleInterface -> Maybe EnvEntry
+lookupExportedEntry scope qname moduleInterface =
+  if (scope, qname) `elem` Set.toList (interfaceExports moduleInterface)
+    then find ((qname ==) . entryName)
+              (Set.toList $ interfaceEntries moduleInterface)
+    else Nothing
```

Comment:
I think you can remove this function in favor of `lookupExportedEntry'`. The check whether the entry is in the `inferfaceExports` is not needed for any test case.